src/OpenColorIO/fileformats/FileFormatSpi3D.cpp

```diff
@@ -172,8 +174,22 @@ CachedFileRcPtr LocalFileFormat::read(
             lutArray[index+0] = redValue;
             lutArray[index+1] = greenValue;
             lutArray[index+2] = blueValue;
-
-            entriesRemaining--;
+            if (ifDuplicated[index] == false)
+            {
+                entriesRemaining--;
+                ifDuplicated[index] = true;
+            }
+            else
+            {
+                std::ostringstream os;
+                os << "Error parsing .spi3d file (";
+                os << fileName;
+                os << "). ";
+                os << "Detect duplicated indices:";
```

Comment:
This message reads a bit oddly, and indicates the calculated index, which would be less helpful for debugging:
```
Detect duplicated indices: <index> duplicates a previous index.
```
It would be clearer to provide the actual indices contained in the LUT file. How about:
```
Data is invalid. A LUT entry is specified multiple times (<rIndex> <gIndex> <bIndex>).
```
See the existing invalid index error for an example of this formatting.

Reply:
I have changed all the points! Thanks for your suggestions!